hypothesis-python/src/hypothesis/internal/filtering.py

```diff
@@ -142,6 +162,8 @@ def merge_preds(*con_predicates: ConstructivePredicate) -> ConstructivePredicate
         del base["exclude_max"]
         if base["max_value"] == math.inf:
             del base["max_value"]
+    if not base["len_func"]:
+        del base["len_func"]
```

Comment:
```suggestion
```
...which makes this redundant!

Reply:
Is this redundant in `base` because the `False` value will be ignored (I thought it would be a cleaner output)? Or for some other reason?